fontforge/svg.c

```diff
@@ -770,26 +770,55 @@ return( SFGetChar(sc->parent,-1,pst->u.subs.variant));
 return( NULL );
 }
 
+#define LIGATURE_MAXLEN 150
+
+char *GetLigature(SplineChar *sc) {
+    PST *best = HasLigature(sc);
+    if ( best != NULL ) {
+    	int32 univals[50];
+        char buf[LIGATURE_MAXLEN] = "";
+        int i, c, r, stored, remain;
+        stored = 0;
+		c = LigCnt(sc->parent,best,univals,sizeof(univals)/sizeof(univals[0]));
+
+		for ( i=0; i<c; ++i ) {
+		    remain = LIGATURE_MAXLEN - stored;
+		    if ( (univals[i]>='A' && univals[i]<='Z') || (univals[i]>='a' && univals[i]<='z') || (univals[i] >= '0' && univals[i] <= '9') || univals[i] == '_' || univals[i] == '-') {
+		        r = snprintf( buf + stored, remain, "%c", univals[i] );
+		    } else {
+                r = snprintf( buf + stored, remain, "&#x%x;", (unsigned int) univals[i] );
+		    }
+
+            if ( r < 0 || r >= remain) {
+               if ((r - remain) > 0) {
+       		       snprintf( buf + stored, remain, "");
```

Comment:
This is a little strange. If the goal here to ensure zero-termination as if the last `snprintf()` hadn't been run it would be clearer to just do that directly, as in `*(buf + stored) = 0;` .